extensions/azure-keyvault/deployment/src/main/java/io/quarkiverse/azure/keyvault/secret/deployment/KeyVaultSecretProcessor.java

```diff
@@ -0,0 +1,106 @@
+package io.quarkiverse.azure.keyvault.secret.deployment;
+
+import java.util.Collection;
+import java.util.List;
+
+import org.jboss.jandex.ClassInfo;
+import org.jboss.jandex.DotName;
+import org.jboss.jandex.Type;
+
+import com.azure.json.JsonSerializable;
+
+import io.quarkiverse.azure.keyvault.secret.runtime.KeyVaultSecretClientProducer;
+import io.quarkus.arc.deployment.AdditionalBeanBuildItem;
+import io.quarkus.deployment.annotations.BuildProducer;
+import io.quarkus.deployment.annotations.BuildStep;
+import io.quarkus.deployment.builditem.CombinedIndexBuildItem;
+import io.quarkus.deployment.builditem.ExtensionSslNativeSupportBuildItem;
+import io.quarkus.deployment.builditem.FeatureBuildItem;
+import io.quarkus.deployment.builditem.IndexDependencyBuildItem;
+import io.quarkus.deployment.builditem.nativeimage.NativeImageProxyDefinitionBuildItem;
+import io.quarkus.deployment.builditem.nativeimage.ReflectiveClassBuildItem;
+import io.quarkus.deployment.builditem.nativeimage.ReflectiveHierarchyBuildItem;
+import io.quarkus.logging.Log;
+
+public class KeyVaultSecretProcessor {
+
+    static final String FEATURE = "azure-keyvault-secret";
+    private static final DotName JSON_SERIALIZABLE_DOT_NAME = DotName.createSimple(JsonSerializable.class.getName());
+
+    @BuildStep
+    FeatureBuildItem feature() {
+        return new FeatureBuildItem(FEATURE);
+    }
+
+    @BuildStep
+    AdditionalBeanBuildItem producer() {
+        return new AdditionalBeanBuildItem(KeyVaultSecretClientProducer.class);
+    }
+
+    @BuildStep
+    ExtensionSslNativeSupportBuildItem activateSslNativeSupport() {
+        return new ExtensionSslNativeSupportBuildItem(FEATURE);
+    }
+
+    @BuildStep
+    IndexDependencyBuildItem indexDependency() {
+        return new IndexDependencyBuildItem("com.azure", "azure-security-keyvault-secrets");
+    }
+
+    @BuildStep
+    void reflectiveClasses(CombinedIndexBuildItem combinedIndexBuildItem,
+            BuildProducer<ReflectiveClassBuildItem> reflectiveClasses) {
+
+        String[] modelClasses = combinedIndexBuildItem
+                .getIndex()
+                .getKnownClasses()
+                .stream()
+                .map(ClassInfo::name)
+                .map(DotName::toString)
+                .filter(n -> n.startsWith("com.azure.security.keyvault.secrets.models."))
+                .sorted()
```

Comment:
By loading the classes in this way, we are creating an implicit contract: All of the classes that start with this package are "safe" to load in this way. Do we know this will always be true?

Reply:
See https://github.com/quarkiverse/quarkus-azure-services/pull/218/commits/5fce42ce138dd83c3e4baf06b1a1b76ec8c82829.